internal/fields/mappings.go

```diff
@@ -624,22 +687,295 @@ func (v *MappingValidator) validateMappingsNotInPreview(currentPath string, chil
+				// logger.Debugf("> Check unmatch: %q (key %q)", currentPath, name)
+				values, err := parseSetting(value)
+				if err != nil {
+					return fmt.Errorf("failed to check unmatch setting: %w", err)
+				}
+				if slices.Contains(values, name) {
+					matched = false
+					break
+				}
+
+				var matches bool
+				if fullRegex {
+					matches, err = stringMatchesPatterns(values, name)
+				} else {
+					matches, err = stringMatchesWildcards(values, name)
+				}
+				if err != nil {
+					return fmt.Errorf("failed to parse dynamic template %s: %w", templateName, err)
+				}
+
+				if matches {
+					// logger.Debugf(">> Issue: matches")
+					matched = false
+				}
+			case "path_match":
+				// logger.Debugf("> Check path_match: %q", currentPath)
+				values, err := parseSetting(value)
+				if err != nil {
+					return fmt.Errorf("failed to check path_match setting: %w", err)
+				}
+				matches, err := stringMatchesWildcards(values, currentPath)
+				if err != nil {
+					return fmt.Errorf("failed to parse dynamic template %s: %w", templateName, err)
+				}
+				if !matches {
+					// logger.Debugf(">> Issue: not matches")
+					matched = false
+				}
+			case "path_unmatch":
+				// logger.Debugf("> Check path_unmatch: %q", currentPath)
+				values, err := parseSetting(value)
+				if err != nil {
+					return fmt.Errorf("failed to check path_unmatch setting: %w", err)
+				}
+				matches, err := stringMatchesWildcards(values, currentPath)
+				if err != nil {
+					return fmt.Errorf("failed to parse dynamic template %s: %w", templateName, err)
+				}
+				if matches {
+					// logger.Debugf(">> Issue: matches")
+					matched = false
+				}
+			case "match_mapping_type", "unmatch_mapping_type":
+				// Do nothing
+				// These comparisons are done with the original data, and it's likely that the
+				// resulting mapping does not have the same type since it could change by the `mapping` field
+				// case "match_mapping_type":
+				// 	logger.Debugf("> Check match_mapping_type: %q (type %s)", currentPath, fieldType)
+				// 	values, err := parseSetting(value)
+				// 	if err != nil {
+				// 		return fmt.Errorf("failed to check match_mapping_type setting: %w", err)
+				// 	}
+				// 	logger.Debugf(">> Comparing to values: %s", values)
+				// 	if slices.Contains(values, "*") {
+				// 		continue
+				// 	}
+				// 	if !slices.Contains(values, fieldType) {
+				// 		logger.Debugf(">> Issue: not matches")
+				// 		matched = false
+				// 	}
+				// case "unmatch_mapping_type":
+				// 	logger.Debugf("> Check unmatch_mapping_type: %q (type %s)", currentPath, fieldType)
+				// 	values, err := parseSetting(value)
+				// 	if err != nil {
+				// 		return fmt.Errorf("failed to check unmatch_mapping_type setting: %w", err)
+				// 	}
+				// 	logger.Debugf(">> Comparing to values: %s", values)
+				// 	if slices.Contains(values, fieldType) {
+				// 		logger.Debugf(">> Issue: matches")
+				// 		matched = false
+				// 	}
```

Comment:
Looks like that these options would be better to skip them, since they would require to have the original type before the documents are stored into Elasticsearch.

If there is no objection, this whole case could be left as empty (with a comment).

Reply:
Yes, I think we can ignore this, but we should still check that the mapping type in the template matches with the actual mapping.